iron-overlay-behavior.html

```diff
@@ -453,14 +453,14 @@
      * @protected
      */
     _applyFocus: function() {
-      if (this.opened) {
-        if (!this.noAutoFocus) {
+      if (!this.noAutoFocus) {
+        if (this.opened) {
           this._focusNode.focus();
+        } else {
+          this._focusNode.blur();
+          this._focusedChild = null;
+          this._manager.focusOverlay();
         }
-      } else {
-        this._focusNode.blur();
-        this._focusedChild = null;
-        this._manager.focusOverlay();
```

Comment:
You should still call the `manager.focusOverlay()` if noAutoFocus is true and opened is false, otherwise you break the flow.